packages/venia-ui/lib/RootComponents/Product/product.js

```diff
@@ -30,7 +30,7 @@ const Product = () => {
     const { error, loading, product } = talonProps;
 
     if (loading && !product) return fullPageLoadingIndicator;
-    if (error) return <div>Data Fetch Error</div>;
+    if (error && !product) return <div>Data Fetch Error</div>;
```

Comment:
Nice find, thanks!

Reply:
I added 3 scenarios for tests
1. when only `product` data exist,
2. when `product` and `error` data exist
3. when only `error` data  exist